Canary publish atom feed is failing

When we merge to main, that triggers the canary build. We do a build and then upload the binaries to a moving tag/release named "canary". We also update the exec mixin entry in the mixins atom feed.
When I merged release/v1 into main, the atom.xml feed update broke. 

This really doesn't actually cause problems other than the build appears to be failed. Let's figure out why this keeps happening and fix it (or disable updating the feed for canary, that may be an option too).

```
2022-10-10T17:10:29.3315938Z Cloning into 'bin/mixins/.packages'...
2022-10-10T17:10:29.6303206Z On branch main
2022-10-10T17:10:29.6310845Z Your branch is up to date with 'origin/main'.
2022-10-10T17:10:29.6311430Z 
2022-10-10T17:10:29.6311903Z nothing to commit, working tree clean
2022-10-10T17:10:29.6314243Z Error: running "git -c user.name='Porter Bot' -c user.email=bot@porter.sh commit --signoff -am Add exec@v1.0.0-47-gb99af468 to mixin feed" failed with exit code 1
```

<details>
<summary>full logs</summary
<code>
2022-10-10T17:10:16.0398019Z ##[section]Starting: Publish Porter Binaries
2022-10-10T17:10:16.0404895Z ==============================================================================
2022-10-10T17:10:16.0405174Z Task         : Command line
2022-10-10T17:10:16.0405448Z Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
2022-10-10T17:10:16.0405836Z Version      : 2.201.1
2022-10-10T17:10:16.0406038Z Author       : Microsoft Corporation
2022-10-10T17:10:16.0406316Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
2022-10-10T17:10:16.0406641Z ==============================================================================
2022-10-10T17:10:16.1733324Z Generating script.
2022-10-10T17:10:16.1743879Z Script contents:
2022-10-10T17:10:16.1744141Z mage PublishPorter PublishMixins
2022-10-10T17:10:16.1745469Z ========================== Starting Command Output ===========================
2022-10-10T17:10:16.1774623Z [command]/usr/bin/bash --noprofile --norc /home/vsts/work/_temp/7ab6c9af-9382-4d6a-b005-4ef7cb1a6b81.sh
2022-10-10T17:10:21.5333980Z bin:
2022-10-10T17:10:21.5334516Z dev
2022-10-10T17:10:21.5334907Z mixins
2022-10-10T17:10:21.5335295Z porter
2022-10-10T17:10:21.5335662Z runtimes
2022-10-10T17:10:21.5336436Z v1.0.0-47-gb99af468
2022-10-10T17:10:21.5336802Z 
2022-10-10T17:10:21.5337202Z bin/dev:
2022-10-10T17:10:21.5337751Z agent-linux-amd64
2022-10-10T17:10:21.5338318Z porter-darwin-amd64
2022-10-10T17:10:21.5338887Z porter-linux-amd64
2022-10-10T17:10:21.5339472Z porter-windows-amd64.exe
2022-10-10T17:10:21.5339779Z 
2022-10-10T17:10:21.5340153Z bin/mixins:
2022-10-10T17:10:21.5340527Z exec
2022-10-10T17:10:21.5340784Z 
2022-10-10T17:10:21.5341154Z bin/mixins/exec:
2022-10-10T17:10:21.5341539Z dev
2022-10-10T17:10:21.5341901Z exec
2022-10-10T17:10:21.5342261Z runtimes
2022-10-10T17:10:21.5342796Z v1.0.0-47-gb99af468
2022-10-10T17:10:21.5343089Z 
2022-10-10T17:10:21.5343476Z bin/mixins/exec/dev:
2022-10-10T17:10:21.5344014Z exec-darwin-amd64
2022-10-10T17:10:21.5344588Z exec-linux-amd64
2022-10-10T17:10:21.5345156Z exec-windows-amd64.exe
2022-10-10T17:10:21.5345453Z 
2022-10-10T17:10:21.5345845Z bin/mixins/exec/runtimes:
2022-10-10T17:10:21.5346377Z exec-runtime
2022-10-10T17:10:21.5346666Z 
2022-10-10T17:10:21.5347225Z bin/mixins/exec/v1.0.0-47-gb99af468:
2022-10-10T17:10:21.5347833Z exec-darwin-amd64
2022-10-10T17:10:21.5348399Z exec-linux-amd64
2022-10-10T17:10:21.5348958Z exec-windows-amd64.exe
2022-10-10T17:10:21.5349265Z 
2022-10-10T17:10:21.5350451Z bin/runtimes:
2022-10-10T17:10:21.5351062Z porter-runtime
2022-10-10T17:10:21.5351304Z 
2022-10-10T17:10:21.5351767Z bin/v1.0.0-47-gb99af468:
2022-10-10T17:10:21.5352273Z agent-linux-amd64
2022-10-10T17:10:21.5352765Z porter-darwin-amd64
2022-10-10T17:10:21.5353249Z porter-linux-amd64
2022-10-10T17:10:21.5353753Z porter-windows-amd64.exe
2022-10-10T17:10:21.5400470Z Updated tag 'canary' (was 3bf49c39)
2022-10-10T17:10:22.2751040Z To https://github.com/getporter/porter.git
2022-10-10T17:10:22.2752323Z  + 3bf49c39...b99af468 canary -> canary (forced update)
2022-10-10T17:10:29.3297825Z Skipping publish binaries for not tagged release v1.0.0-47-gb99af468
2022-10-10T17:10:29.3315938Z Cloning into 'bin/mixins/.packages'...
2022-10-10T17:10:29.6303206Z On branch main
2022-10-10T17:10:29.6310845Z Your branch is up to date with 'origin/main'.
2022-10-10T17:10:29.6311430Z 
2022-10-10T17:10:29.6311903Z nothing to commit, working tree clean
2022-10-10T17:10:29.6314243Z Error: running "git -c user.name='Porter Bot' -c user.email=bot@porter.sh commit --signoff -am Add exec@v1.0.0-47-gb99af468 to mixin feed" failed with exit code 1
2022-10-10T17:10:29.6375638Z ##[error]Bash exited with code '1'.
2022-10-10T17:10:29.6417961Z ##[section]Finishing: Publish Porter Binaries

</code>
</details>

--- a/magefile.go
+++ b/magefile.go
@@ -97,11 +97,13 @@ func XBuildAll() {
 func XBuildPorter() {
 	mg.Deps(copySchema)
 	releases.XBuildAll(PKG, "porter", "bin")
+	releases.PrepareMixinForPublish("porter")
 }
 
 // Cross-compile the exec mixin
 func XBuildMixins() {
 	releases.XBuildAll(PKG, "exec", "bin/mixins/exec")
+	releases.PrepareMixinForPublish("exec")
 }
 
 // Generate cli documentation for the website

